src/main/java/co/alphavantage/output/exchange/CurrencyExchange.java

```diff
@@ -0,0 +1,77 @@
+package co.alphavantage.output.exchange;
+
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import co.alphavantage.output.AlphaVantageException;
+import co.alphavantage.output.JsonParser;
+import co.alphavantage.output.exchange.data.CurrencyExchangeData;
+
+public class CurrencyExchange {
+	private CurrencyExchangeData data;
+	
+	CurrencyExchange(CurrencyExchangeData data){
+		this.data = data;
+	}
+	
+
+	public CurrencyExchangeData getData() {
+		return data;
+	}
+
+
+  /**
+   * Create Currency Exchange data representation from json object
+   * @param json string to parse
+   * @return Currency Exchange data
+   * @throws AlphaVantageException
+   */
+	public static CurrencyExchange from(String json) throws AlphaVantageException {
+		CurrencyExchangeParser parser = new CurrencyExchangeParser();
+		return parser.parseJson(json);
+	}
+
+  /**
+   * Helper class for parsing json to {@code Currency Exchange}
+   */
+	private static class CurrencyExchangeParser extends JsonParser<CurrencyExchange>{
+		public CurrencyExchange resolve(JsonObject rootObject) {
+			  Type dataType = new TypeToken<Map<String, Map<String, String>>>() {
+		      }.getType();
+		      try {
+		        Map<String, Map<String, String>> data = GSON.fromJson(rootObject, dataType);
+		        List<CurrencyExchangeData> exchanges = new ArrayList<>();

```

Comment:
What about this to prevent from creating and fill an arraylist from which we will only get the first value?

<img width="711" alt="captura de pantalla 2017-12-10 a les 23 26 29" src="https://user-images.githubusercontent.com/1654802/33810134-9bae1e10-de01-11e7-908d-123512c26f63.png">

Reply:
I just added the code to multiples keys (if we have in the future this option) but there is no problem I agree with you.